nni/compression/pytorch/speedup/infer_shape.py

```diff
@@ -273,7 +273,8 @@ def __repr__(self):
     'aten::mean': lambda module_masks, mask, shape: mean_inshape(module_masks, mask, shape),
     'Dropout': lambda module_masks, mask: dropout_inshape(module_masks, mask),
     'Dropout2d': lambda module_masks, mask: dropout_inshape(module_masks, mask),
-    'aten::dropout': lambda module_masks, mask: dropout_inshape(module_masks, mask)
+    'aten::dropout': lambda module_masks, mask: dropout_inshape(module_masks, mask),
+    'aten::detach': lambda module_masks, mask: dropout_inshape(module_masks, mask)
```

Comment:
Why detach should behave like a dropout OP?

Reply:
Detach only affects the behavior in gradient backpropagation. And sparsity should be exactly the same between input and output tensor. I found this logic is already implemented in dropout callback so I directly put it here.